crates/walrus-service/src/node/node_recovery.rs

```diff
@@ -215,25 +242,18 @@ impl NodeRecoveryHandler {
     }
 
     /// Restarts any in progress recovery.
-    pub async fn restart_recovery(&self) -> Result<(), TypedStoreError> {
+    pub async fn restart_recovery(&self) -> anyhow::Result<()> {
         if let NodeStatus::RecoveryInProgress(recovering_epoch) = self.node.storage.node_status()? {
-            if recovering_epoch == self.node.current_epoch() {
-                // The `latest_event_epoch` is still set to `0` at this point.
-                self.node
-                    .latest_event_epoch
-                    .store(recovering_epoch, Ordering::SeqCst);
-                return self.start_node_recovery(self.node.current_epoch()).await;
-            } else {
-                assert!(recovering_epoch < self.node.current_epoch());
-                tracing::warn!(
-                    recovering_epoch,
-                    current_epoch = self.node.current_epoch(),
-                    "recovery epoch mismatch; skip recovery restart; next epoch change start event \
-                    will bring node to the latest state"
-                );
-                self.node.set_node_status(NodeStatus::RecoveryCatchUp)?;
-            }
+            tracing::info!("waiting for latest event epoch to be set to restart node recovery");
+            self.node.current_event_epoch().await?;
```

Comment:
I don't know we actually need to wait for the current_event_epoch() to be populated, since any caller of current_event_epoch() now waits for it. However, since this is the previous behavior, let's keep it to be safe. This is also an one of operation during node start up, so it doesn't have much overhead.